src/App.tsx

```diff
@@ -15,36 +17,109 @@
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+const SORT_FIELD_ALPHABET: string = 'alphabet';
+const SORT_FIELD_LENGTH: string = 'length';
+const SORT_FIELD_RESET: string = '';
+
+type Props = {
+  sortField: string;
+  setSortField: () => string;
+  isReverse: string;
+  setReverse: () => string;
+  visibleGoods: string[];
+};
+
+function sortGoods(
+  goods: string[],
+  sortField: string = '',
+  isReverse: boolean = false,
+): string[] {
+  const visibleGoods: string[] = [...goods];
+
+  if (sortField) {
+    switch (sortField) {
+      case SORT_FIELD_ALPHABET:
+        visibleGoods.sort((good1, good2) => good1.localeCompare(good2));
+        break;
+      case SORT_FIELD_LENGTH:
+        visibleGoods.sort((good1, good2) => good1.length - good2.length);
+        break;
+      default:
+        throw new Error('Sort method is unknown!');
+    }
+  }
+
+  if (isReverse) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
+export const App: React.FC<Props> = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReverse, setReverse] = useState(false);
+  const visibleGoods = sortGoods(goodsFromServer, sortField, isReverse);
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={classNames('button is-info', {
+            'is-light': sortField !== SORT_FIELD_ALPHABET,
+          })}
+          onClick={() => {
+            setSortField(SORT_FIELD_ALPHABET);
+          }}
```

Comment:
[CODE STYLE] - The function set within `onClick` should start with a verb to indicate its action. Consider renaming the inline function to something like `handleSortAlphabetically`.